Src/FluentAssertions/Common/PropertyInfoExtensions.cs

```diff
@@ -9,4 +9,16 @@ internal static bool IsVirtual(this PropertyInfo property)
         MethodInfo methodInfo = property.GetGetMethod(nonPublic: true) ?? property.GetSetMethod(nonPublic: true);
         return !methodInfo.IsNonVirtual();
     }
+
+    internal static bool IsStatic(this PropertyInfo property)
+    {
+        MethodInfo methodInfo = property.GetGetMethod(nonPublic: true);
```

Comment:
The removed code in 2007fa89db528c4cdfe2eb50d7cba2e0108c00a6 was correct, there just wasn't any unit tests using it.

It's quite nice, but it's valid to write a setter-only property
```cs
public class MyClass
{
    public static bool MyProperty { set; }
}
```

without ` ?? property.GetSetMethod(nonPublic: true)` `methodInfo` will now be `null` for a setter-only property and then invoking `IsStatic` on it throws a `NullReferenceException`.